repos/system_upgrade/el8toel9/actors/networkdeprecations/tests/unit_test_networkdeprecations.py

```diff
@@ -1,148 +1,125 @@
-from leapp.libraries.common.testutils import create_report_mocked, make_OSError
-
-
-def _listdir_nm_conn(path):
-    if path == networkdeprecations.NM_CONN_DIR:
-        return ['connection']
-    raise make_OSError(errno.ENOENT)
-
-
-def _listdir_ifcfg(path):
-    if path == networkdeprecations.SYSCONFIG_DIR:
-        return ['ifcfg-wireless']
-    raise make_OSError(errno.ENOENT)
-
-
-def _listdir_keys(path):
-    if path == networkdeprecations.SYSCONFIG_DIR:
-        return ['keys-wireless']
-    raise make_OSError(errno.ENOENT)
-
-
-def test_no_conf(monkeypatch):
+from leapp.models import (
+    IfCfg,
+    IfCfgProperty,
+    NetworkManagerConnection,
+    NetworkManagerConnectionProperty,
+    NetworkManagerConnectionSetting
+)
+from leapp.reporting import Report
+from leapp.utils.report import is_inhibitor
+
+
+def test_no_conf(current_actor_context):
     """
     No report if there are no networks.
     """
 
-    monkeypatch.setattr(networkdeprecations.os, 'listdir', lambda _: ())
-    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
-    networkdeprecations.process()
-    assert not reporting.create_report.called
+    current_actor_context.run()
+    assert not current_actor_context.consume(Report)
 
 
-def test_no_wireless(monkeypatch):
+def test_no_wireless(current_actor_context):
     """
     No report if there's a keyfile, but it's not for a wireless connection.
     """
 
-    mock_config = mock.mock_open(read_data='[connection]')
-    with mock.patch('builtins.open' if six.PY3 else '__builtin__.open', mock_config):
-        monkeypatch.setattr(networkdeprecations.os, 'listdir', _listdir_nm_conn)
-        monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
-        networkdeprecations.process()
-        assert not reporting.create_report.called
+    not_wifi_nm_conn = NetworkManagerConnection(filename='/NM/wlan0.nmconn', settings=(
+        NetworkManagerConnectionSetting(name='connection'),
+    ))
 
+    current_actor_context.feed(not_wifi_nm_conn)
+    current_actor_context.run()
+    assert not current_actor_context.consume(Report)
 
-def test_keyfile_static_wep(monkeypatch):
+
+def test_keyfile_static_wep(current_actor_context):
     """
     Report if there's a static WEP keyfile.
     """
 
-    STATIC_WEP_CONN = textwrap.dedent("""
-        [wifi-security]
-        auth-alg=open
-        key-mgmt=none
-        wep-key-type=1
-        wep-key0=abcde
-    """)
+    static_wep_nm_conn = NetworkManagerConnection(filename='/NM/wlan0.nmconn', settings=(
+        NetworkManagerConnectionSetting(name='wifi-security', properties=(
+            NetworkManagerConnectionProperty(name='auth-alg', value='open'),
+            NetworkManagerConnectionProperty(name='key-mgmt', value='none'),
+            NetworkManagerConnectionProperty(name='wep-key-type', value='1'),
+            NetworkManagerConnectionProperty(name='wep-key0', value='abcde'),
```

Comment:
@lkundrak so we discussed the secrets solution and for now, implement an allowlist of keys for which you need (and store) the value and for the others set None. We will investigate a better solution around secrets for the future but right now let's go just in this way.